File: examples/c++/aio.cpp
/*
 * Author: Brendan Le Foll <brendan.le.foll@intel.com>
 * Contributors: Alex Tereschenko <alex.mkrs@gmail.com>
 * Contributors: Manivannan Sadhasivam <manivannan.sadhasivam@linaro.org>
 * Copyright (c) 2014 Intel Corporation.
 *
 * SPDX-License-Identifier: MIT
 *
 * Example usage: Reads ADC A0 value continuously. Press Ctrl+C to exit.
 */

/* standard headers */
#include <iostream>
#include <signal.h>
#include <stdlib.h>

/* mraa headers */
#include "mraa/aio.hpp"
#include "mraa/common.hpp"

/* AIO port */
#define AIO_PORT 0

volatile sig_atomic_t flag = 1;

void
sig_handler(int signum)
{
    if (signum == SIGINT) {
        std::cout << "Exiting..." << std::endl;
        flag = 0;
    }
}

int
main(void)
{
    uint16_t value;
    float float_value;

    signal(SIGINT, sig_handler);

    //! [Interesting]
    /* initialize AIO */
    mraa::Aio aio(AIO_PORT);

    while (flag) {
        value = aio.read();
        float_value = aio.readFloat();
        std::cout << "ADC A0 read %X - %d" << value << value << std::endl;
        std::cout << "ADC A0 read float - %.5f" << float_value << std::endl;
    }
    //! [Interesting]

    return EXIT_SUCCESS;
}

File: examples/python/aio.py
#!/usr/bin/env python

# Author: Brendan Le Foll <brendan.le.foll@intel.com>
# Copyright (c) 2014 Intel Corporation.
#
# SPDX-License-Identifier: MIT
#
# Example Usage: Reads integer and float value from ADC

import mraa

print(mraa.getVersion())

try:
    # initialise AIO
    x = mraa.Aio(0)

    # read integer value
    print(x.read())

    # read float value
    print("%.5f" % x.readFloat())
except:
    print("Are you sure you have an ADC?")
